### DISCUSSION TITLE
Mining rewards based on Quadratic staking

### DISCUSSION THREAD

**Discussion by midenaio** [idena-network/idena-docs#81] [IIPs]
We propose to change the block reward distribution so that instead of a fixed mining reward, the block proposer and block validators are paid in proportion to their stake to the power of 0.9. The proposed change affects only mining rewards distribution and does not affect the consensus mechanism, which implies equality of votes when verifying blocks according to the principle: one node = one vote.

![image](https://user-images.githubusercontent.com/47352542/178466020-a6e05081-ea29-4a6d-b48e-d46bc9ddd67a.png)

**Comment by Zen-44** [comment]
Wouldn't this discourage members with small stakes (new members for example) from mining at all? Even if the voting power is kept the same, there will be a small incentive for identities with low stakes. Having the same voting power won't be as useful if users are not incentivized the same to use it. Some may buy more coins to earn higher mining rewards, but for new people that won't buy coins, mining will become really unprofitable.

**Comment by midenaio** [comment]
@Zen-44 thank you for the input. We have added the modeling results of the mining rewards before and after IIP-5.
The quadratic nature of the mining rewards gives a higher yield for the smaller stakes. This means that the mining income of the validators with a small stake is not affected significantly.

**Reply by ubiubi18** [reply]
I wonder if the quadratic nature of this proposal is strong enough? Is it possible to apply something closer to "p=0.5" this time instead of "p=0.9"?
Let us ask ourself what is worse: Allow human farms to reach higher profitablity by spreading their stake on newbie accounts? Or making it harder for pools to do that, but also harder for late adopter solo miners to become profitable with small stakes in the first place? A unique selling point of idena is the cheap accessability for late adopters so far. Everywhere else you have to invest a lot of (fiat) coins first before you can participate profitably in mining or staking. In idena you had so far to invest only time and something like 5 to 7 dollar worth Computerconnection a month to access the network properly. With IIP5 this precondition to invest just a small ammount of time and money will grow quadratically? kind of forcing late adopters to either invest many month of unprofitable accumulating stake, or to invest a growing ammount of fiat money right away to become profitable in the first place?

RIght now it may only be an ammount like 600 idna, which equals 20$ of worth. But over time the percentage of staked coins will grow just like the price. So i can realistically imagine a close future where you need five times more staked coins which you need to buy at a five times higher price or which take five times longer to mine. At that point we would talk about thousands of $ allready - and that would for sure force most late adopters to join pools or invest thousands of $ or miss idena out.

This might all be desirable from the perspective of attracting investors, but what about the idea of "democratic mining"?

Taking all this into account I think the argument you made against "p=0.5" in IIP4 is not that strong anymore. So i would ask you to consider adjusting "p=0.9" a bit more in favor of late adopters, even if that lowers the profitabilitiy of solo miner stake investors. Big Investments would be still possible in a "p=0.5" world, they would just need to chose the route of investing in mining pools, which is maybe technically more demanding. 

It is for sure at this point in time easier for new investors to ape in into p=0.9 staking, but in the long run the downside of making profitability harder for late adopter solo miners could be more dramatic?

If IIP5 was mainly concerned about reducing influence of human farms, than i would argue there are probably more straight forward ways to achive that, like making flip production rewards time crittical, like reward only flips that where produced within the first 1000 blocks of each new epoch, so that centralised flip production over three weeks in a row is made unprofitable in favor of involving more people into flip production, which would grow production costs and theirby reduce centralised influence for farms substancially.

**Reply by midenaio** [reply]
@ubiubi18 thanks for your suggestion. We’ve run another model (data is slightly different because of some randomness). You can see the difference between the mining rewards for the Large pools before and after IIP-5 with Stake^0.9 and Stake^0.5. 


![image](https://user-images.githubusercontent.com/47352542/178785940-2f648758-b40a-4f5a-ac13-02e568cd6cf9.png)


With the capital discriminator 0.5 large pools still get big advantages in mining rewards regardless of the fact that they do not replenish their stakes. It happens because with 0.5 the stake plays a smaller role than the number of accounts. We think that currently 0.9 is a kind of a compromise though it can be changed in the future if the big stakeholders significantly dominate the mining.

**Reply by ubiubi18** [reply]
In most blockchain projects mining rewards and costs of running a mining node are allways pretty close to each other. That will likely happen in idena too. So it would be bad for late adoption in the long run if it would take eventually years for late adopters before running a mining node becomes a positive sum game. I think it would be nice if you could add a mechanism that allows late adopters to join the positive sum game within less than a few dozen epochs without the need for fiat-investments. Maybe something like that could work (?): https://discordapp.com/channels/634481767352369162/647573315640098817/998496101319594034

**Comment by reb0rn21** [comment]
I welcome the change.......
power is in balance and making "bed actors" have less initiative

**Comment by 0JustinMiles0** [comment]
I'm also concerned on the negative impact IIP-5 would have on the future growth of the network as mining revenue would depend on an initial capital investment for newcomers. More concerning, the initial capital investment that would be required for newcomers would increase as the network grow making it more expensive over time to make an earning from mining. I think this proposal conflicts with the fundational principle of "democratic mining" as described in the Idena whitepaper:

_Idena enables democratic access to mining: Neither expensive mining hardware **nor a bunch of money for stake is needed**, but rather an average laptop that is online._

This proposal modifies the economic incentives of the consensus mechanism as such, I do not agree that this proposal does not affect the consensus mechanism since the economic incentives are an integrate part of the consensus mechanism. Post IIP-5, the consensus mechanism wouldn't be purely PoP (Proof-of-Personhood) but an hybrid between PoP and PoS (Proof-of-Stake).

This proposal needs to document the impact on democratic access to mining rewards over time for anyone who can prove its personhood. Particularly, we should assess the amount of initial capital that will be required to stake by newcomers as the network grow in order to earn as many iDNAs as they can earn today with the current reward mining distribution.

**Reply by reb0rn21** [reply]
Democratic access ≠ pool paid human bot

But I do agree we need to address newcomers initiative more....... while it would not be easy abused by pool operators

**Reply by midenaio** [reply]
Idena is aimed to be the most decentralized network of nodes based on the principle of one person=one node. Due to excessive decentralization and an excessive number of miners, the problem of blockchain scalability is to be solved. A large number of online nodes will be divided into segments (shards) and transactions will be performed asynchronously (Sharding 2.0).

However, we see that despite the principle of one person=one vote, with the growth of the network, the number of nodes does not increase, but falls. This is due to the dominance of pools, which leads to an excessive concentration of mining rewards and demotivates solo miners to keep their nodes. There are 7998 accounts represented by only  38 nodes (large pools). Independent miners cannot compete with pools and simply shut down their nodes. We believe that the growth of the network associated with the growth of pools is contrary to our goal and interferes with the solution of the scalability trilemma.

With quadratic mining rewards, several hundred sleeping nodes can be back in the game. Due  to the stake discrimination (stake^0.9), we will provide resistance to the capture of the network by large stakeholders. With the dominance of large stakeholders, the degree can be revised by the network in future.

**Reply by 0JustinMiles0** [reply]
@midenaio thanks for taking the time to comment. 

Are you comfortable moving forward with IIP-5 without assessing its impact on mining rewards distribution (how much iDNAs they can expect to earn from mining) over time for newcomers?

At minimum, this effect should be mentioned in the proposal as a tradeoff many people voiced their concern about. It is even admitted in the proposal that there is a risk (_"If the proposed changes cause an unwanted imbalance in the economics of iDNA"_) but it should be addressed more specifically so voters can make a decision with full knowledge of the facts.

**Reply by reb0rn21** [reply]
You know priority is blockchain security do kid of age 3 have income, no they do not
pools are flooding and try to spam with new accounts etc, should they earn for it, I hope they do not

Do I want more newbies that are really interested in idena, I sure do
Do I want pool newbies. no if you ask me defo not, pools should be targeted even more, time of validation should be changed so it harm pool most, etc ley say CET 9PM

**Comment by Zen-44** [comment]
@midenaio Do you think that after these changes the people starting from scratch without investing external capital for buying coins will be able to mine coins for profits? I understand where this proposal is headed and what it is aiming to solve. I also understand replenishing stakes is an option after you accumulate some coins. 

My concern is that new members with really low stakes (newbies) will not be incentivized enough to invest their time in the project if they can't afford to buy coins to raise their stake. They would need to pass several validations to get some decent amount of coins to stake. I think a new member would simply not even try if they have to invest a lot of their time upfront to be able to mine efficiently (in terms of money) several months after they start. To put it simply, I don't think starting your journey in Idena from scratch would be as user-friendly. From the "Modeling" changes added to IIP, the change in mining rewards don't seem so drastic, but I am not sure how the numbers look for ranges like [0;50] iDNA stakes.

Has it been considered to use the locked coins of new members during the time they are newbies for staking purposes? From what I know, the locked balance is not treated as a stake and it's just frozen coins. I think treating it as a stake would somewhat help new members in keeping up with the rest of the network while not giving them a huge advantage. Most identities don't accumulate more than 100-200 iDNA in their locked balance before they reach Verified status and I think it would encourage people who don't have anything material to invest to start an identity by not letting them fall behind for several epochs before they can start adding some substantial amount of coins to their stakes.

**Reply by midenaio** [reply]
With IIP-5 identities with Newbie status, will get less mining rewards. Newbies need to accumulate their coins across several validations by creating good flips and reporting bad flips.

This is quite aligned with IIP-3 which introduced the discrimination of Newbies. Their votes have no role to confirm blocks. Such discrimination against Newbies looks logical due to the fact that Newbie needs to go through 3 validations to prove their humanness. 

However, despite discrimination, Newbies can still run their nodes and earn from mining while maintaining network connectivity. Newbie’s locked coins are also counted for staking rewards. They can increase their mining rewards by replenishing the stake and accepting the risk of a possible loss during the validation fail.

**Reply by Zen-44** [reply]
It's great to hear locked coins are counted! This way new members still have some way to compete for rewards. The quadratic formula gives a lot of value to locked coins at their early stage and further incentivises them! 👍

**Comment by rioda-org** [comment]
People would actually run their nodes on VPS's again. More nodes, greater decentralisation. Biggest selling point of Idena, exaggerated decentralisation. Best marketing will be blockchain stats.

**Comment by Ed-Relight** [comment]
From the proposal [page](https://docs.idena.io/docs/iip/iip-5#mining-rewards-in-pools):
"If a pool owner is selected as the block proposer, the block reward is determined by the stake of the pool representative. The pool representative is selected from the list of all delegates using the round robin algorithm."

Could you expand on this a bit more?
How often is the representative selected?
Does round robin algorithm imply that a representative will be chosen iteratively from the pool's identities each time/block?

**Reply by Zen-44** [reply]
The round robin selection process goes through a list in a rational order. This means that it would choose all the elements in the group equally. 

Every time a pool is chosen to vote or issue a block, it has to select a representative. Please also note that if a pool's address is validated, it can be selected as a representative itself along with all the delegators.

**Reply by hlolve** [reply]
If a pool node, is restarted, the round robin order in maintained at last state?

**Comment by cryptogeek555** [comment]
I support the changes. It will definitely increase the number of mining nodes as solo mining could become profitable for most identities. Large pools will get less rewards, thus they will dump less or maybe running a pool will  become unprofitable and they quit which is good as most of them parasitizing on the network (creating bad flips, false reporting or no reporting at all).
The demand for idna will increase and it may lead to pumping the price ->the growth of the network with real individuals, not pool workers

**Comment by spiketrains** [comment]
### Abstract

I believe this is a big change to the tokenomics and the long-term consequences are not obvious. **_I don't expect a strong impact on the number of solo miners in the short term._** Nevertheless, at the moment I am still more inclined towards supporting this proposal.

### Effect on large, centralized pools

- I believe large pools will adapt to be as profitable as possible. This is definitely easier for large, centralized "professional" pools in which the pool managers can distribute the stake optimally among the identities. These pools will probably increase their total stake. I appreciate the simulation on the effect of IIP-5, but I have read too many posts in other channels ignoring how the situation can be very different if pools decide to increase their stake.
- These pools normally pay a flat compensation to their members. I don't see a path for considerable change in number of solo miners here. People who agreed on this arrangement won't become solo miners because of this change as they are probably unwilling to (1) set up an independent node and/or (2) acquire a considerable amount of iDNA to put in stake. IIP-4 has already gave us preliminary data about this but it might be too early to extrapolate longer-term effects.

### Effect on staking pools

- I call here staking pools the ones in which the pool operator does not have access to the delegated identities, meaning that it can't distribute the stake optimally. **_In this regard, they will be at a disadvantage w.r.t. large centralized pools as they won't be able to optimize the mining rewards._**
- In this kind of staking pools, people accept to pay a fee to the pool operator who will then distribute mining and validation rewards in function of their contribution to the pool. Even though setting up an Idena Network node is easy and inexpensive, some people prefer paying for minimizing the effort, as small as it might be.
- I can argue that these people won't have any incentive in becoming solo miners as long as (1) the pool operator distributes the rewards according to the quadratic staking principle and (2) they really trust the pool operator. Point (2) is especially meaningful for large stakers. I can see some very large stakers deciding to go "solo".

### Effect on new entries (and dormant identities) with capital

- As detailed in the sections above, I don't expect a big influx of solo miners from pools. We should then think about potential new entries.
- It is clear that the combined effect of IIP-4 and IIP-5 might be enticing for new entries deciding to deploy capital to exploit the high return for large stakers. 
- Dormant identities with large stakes might suddenly find out to be profitable enough to become active.
- However, **_both new entries and dormant identities might easily decide to join staking pools_**. New entries might include people with capital with absolutely no interest in the project principles. They come with money only to make more money. Joining a staking pool with low fees might be a no-brainer for them, especially if we are talking about people involved in N projects rotating capital from chain to chain and from dapp to dapp.
- In conclusion, I see here a reason for an increase in solo miners, but we need to be realistic on how staking pools inevitably reduce decentralization. When we discuss huge changes to tokenomics like this we should also think about the long term. I can imagine staking pools becoming always more attractive, especially with the possibility to deploy and exploit more complex smart contracts.

### Effect on "organic" new entries 
- I call organic new entries people who would have joined Idena Network regardless because they genuinely like the principles and the vision behind the project.
- I strongly believe that most solo miners are in this category. We don't solo mine for higher profitability than joining a staking pool.
- New entries who can't or don't want to deploy a strong starting capital for their stake might **_feel_** at a strong disadvantage. This can be discouraging. It is easy to generate FUD making examples on what happens when you "meet" a block proposer or a bunch of validators with huge stakes. I think we need to be careful here to clarify that there are still components of the rewards that are truly egalitarian.
- I believe that even though stake^0.9-dependent mining rewards puts small stakers at less disadvantage than stake^1 mining rewards, what will dominate here in the "psychology" of the participants is instead the clearly "competitive" rewards distribution. Some people would be more comfortable with a fixed APY (and variable absolute inflation) as it would feel less "competitive", even though they wouldn't be better off! I am **not** suggesting to go in this direction but I want to mention this "psychological" aspect. Let's not underestimate it.

### Effect on tokenomics

- I believe at this point it is good to increase incentives for staking. Maybe it's a bit too early to do it right after IIP-4, but maybe we should have gone in this direction already some time ago and it's not good to wait any longer. 

### Conclusion
I believe it's worth to try. It's a strong change and I am not really optimistic on the increase in number of solo miners, but I still believe it's good to make human identities with high stakes more valuable.

**Reply by 0JustinMiles0** [reply]
Great breakdown! I agree with most of your assessment except that I would prefer we take more time to better assess the different dynamics you laid out before jumping in. The current proposal lacks segmentation of the different type of pools and how each of them will adapt post IIP-5. As you said, I think the impact on identities leaving pools to solo mine is greatly overstated. Farm workers just have little interest in running their own node.

**Reply by midenaio** [reply]
@Spiketr4in We appreciate your feedback on the proposal. As for staking pools, we think they are not be feasible in the network. To get maximum profit a pool should distribute an investor's funds across many "human" identities. If an investor wants to withdraw the money, the pool must kill those "human" identities to be able to return the money.  Then the pool needs to grow new "human" identities by validating them 4 times. So the scheme is quite complicated.

**Comment by hlolve** [comment]
In IIP-4 only 18% of mined coins during validation ceremony are under influence of stake weight. To each block mined, is possible to apply only a percentage of rewards to stake weight ?

**Comment by simontakouridis** [comment]
Here's a better idea that's simpler and more effective:

Have a minimum "mining stake" in order to be able to mine. This mining stake is separate from the iip4 stake and therefore does not garner staking rewards. The mining stake is simply an enabler to be able to mine (of course the miner must also have human status).

In effect, if someone has an amount of iDNA equivalent to the minimum mining stake, it is more profitable to mine. However, if someone has orders more iDNA than the minimum mining stake, then it is more profitable to invoke iip4 staking rewards rather than form a pool.

The advantage of this to the current iip5 proposal is:
- it is very simple to implement.
- it is effective in that pools will opt for iip4 staking rewards.
- the minimum mining stake can be reduced in future to expand the number of validators, e.g. in response to high tx fees.
- it does not have the associated risk of plutocracy, or having so few validators that there is a risk of attack (as with the current iip5 proposal).

A clear and present danger to Idena are farming pools. This simple solution will redirect their profit making to iip4 and thus give Idena time to develop a solid foundation so that it can become more inclusive of people and pools in future.

**Comment by 0JustinMiles0** [comment]
As everybody seems to agree that the higher the share of solo miners the healthier the Idena network is, any proposal attempting to increase the share of solo miners should be considered.

That said, multiples people have expressed their concerns that the current proposal is attempting to achieve this goal at the detriment of a core foundational principle of the Idena network: "democratic mining".

An alternative mechanism to the one proposed in IIP-5 could be to implement a "pool tax". This tax would be taken off the mining reward from pools and redistributed to solo miners. This tax could be higher the larger the mining pool is. Pools who run farm workers may decide to split in smaller pools to reduce the tax but may still wish to manage their identities through pools so that can still benefit from the convenience and economies of scale offered by pools. Regardless on whether those pools would split or not, it is recognized that farm workers aren't generally interested in becoming solo miners. 

The primary mechanism through which the "pool tax" would attempt to increase the share of solo miners would be through the creation of an economic incentive for non-farm workers to solo mine rather than to participate in a pool. The economic incentive to leave pools would increase as the share of solo miners decreases.

Regardless of the parameters of the "pool tax" that would need to be defined and simulated, it appears that this mechanism would not present a risk to the integrity of democratic mining on the Idena network. For this reason, I urge the author of IIP-5 to consider this alternative (and others) before moving forward with IIP-5.

**Comment by hlolve** [comment]
More value for "The vote". Only non delegated IDs can be in committee for approve and receive ads rewards. Farm pools with many accounts controlling IDs keys must be out of reviewing ads. Pools with many IDs can auto approve their ads at present number of IDs, it is supposed that pools only have 1 vote.

**Comment by Ed-Relight** [comment]
@midenaio what do you think about having some UBI value for mining rewards, so that even the identities with virtually no stake can get some guaranteed reward (like [this Newbie](https://scan.idena.io/identity/0x00276CC9bB0f5074bd09CA495A7A0Aa06C5d8c82/epoch/88/rewards), who got mere 5 idna for validation, will have zero motivation to mine, since the rewards would be almost inexistent)

Basically,
if(stake<100)
CalculateMiningRewards(Stake = 100)

Would be interesting to see how this proposal changes the rewards distribution across identities

**Reply by hlolve** [reply]
It will benefits pool farms, killing IDs and starting again

**Reply by Ed-Relight** [reply]
Depends on the stake threshold, identities can earn much more with invitation and flip reward if this threshold is low enough

**Reply by midenaio** [reply]
@RelightMotion Any preferences in mining rewards given to newcomers can be used by pools with a large number of hired people and invites. This leads to a decrease in the income of other participants and a decrease in the number of nodes.

At the same time due to IIP-3 newbies' nodes do not make a significant contribution into the network security so additional stimulation of mining by newcomers is not required.

On the other hand, stimulating mining competition for large stakeholders is aimed at increasing the valuation of the network and increasing the income from mining (denominated in USD) for all network participants, including newcomers.

**Comment by dalikexterminate** [comment]
If in pools the rewards are still calculated in the same way as in solo mining, then why the banquet?  maybe due to the fact that there are many accounts in the pools with a small stake balance, and all the others (solo human and family pools) have more?  so the pools will get less.  now I understood))

this happened because the pools remove fatty stakes.  will now transfer them from Suspendeds to Humans.  and yes, they will sell less coins.  this should work. may be))
